### ISSUE TITLE
rune curses

### ISSUE THREAD

**Issue by jeremygurr** [jeremygurr/dcssca#241]
Since I'll be implementing this soon, it'd be good to discuss it a bit first. 

The rune curse idea is that for each rune picked up, the game changes to become more difficult. So a 3 run win will be a little harder than it was before, but a 5 rune win should be quite a bit harder, and a 15 rune win should be nearly impossible. 15 rune wins should be almost unheard of, at least on normal and hard mode. 

For starters, each rune picked up adds 10% (stackable) to all monsters hp. So:
- 3 runes: +33% hp
- 5 runes: +61% hp
- 10 runes: +159% hp
- 15 runes: +318% hp

that alone would be a bit of a challenge, but it gets worse.

Each rune will have it's own specific "curse". Here are some initial ideas:
- serpentine: stairs randomly teleport nearer monsters ~every 50 turns
- decaying: player rots at a steady rate. 
- barnacled: player movement is slowed down by 0.2
- gossamer: traps more common and more deadly
- abyssal: monster spawning increased significantly. 
- silver: ascending stairs sometimes moves player to a cluster of monsters on same floor instead
- slimy: player randomly mutates, good or bad 50/50, like a bad version of evolve.
- dark: player vision is reduced to 3, but can still sense the position of monsters in normal LOS, just can't id them until they are closer.
- glowing: contamination doesn't auto decay. Can be cured by drinking any potion. 
- fiery: monster damage increased by 25%
- magical: power of monster spells increased
- demonic: when a hit does a lot of damage to a monster, there is a chance that a demon will spawn. 
- golden: gold in los has a chance to distract player, causing their attack to miss, or their spell to fail. Higher chance of gold being dropped by monsters. 
- iron: walls are all indestructable, but paths to stairs are always guaranteed. 
- icy: ice walls or tunnels randomly appear. Can be melted with heat. Never close to player. 
- obsidian: walls and trees adjacent to the player occasionally come to life and attack the player, but don't move. When the player moves away from a living wall, it eventually fades back into a normal wall.
- bone: when a monster dies, there is a 1/3 chance it comes back as a zombie or demon

I'm sure some of these won't work, and will need to be changed. This is just a first stab at it. And some aren't that interesting. I'd ideally like each to change the game in a flavorful and challenging way. 

It's hard to get a game to the point where you can really feel the rune curses. You can use wizard mode to create a level 25 char and pick up some runes, and that would help test some things. But you can't fully replace a real run where you've invested heavily in your character. Maybe people that want to help test this can start in wizard mode, but not use wizard mode to change the character artificially, but just use wizard mode to prevent deaths from a bad rune curse mechanic. 

On the positive side, I can adjust the rune curse behavior even for existing games. So if we find that we need to tweak the silver rune curse, you can save your game, I can apply the changes, then you can reload your game with those changes in effect.

**Comment by Sandman25DCSS** [comment]
This makes caster wins almost impossible (no MP will be enough to kill 318% HP Hell Sentinel, for example) and the easiest combo becomes FoFi of Makhleb in heaviest armour found. Also I dislike some of rune effects, not sure if you want to discuss them.

**Comment by jeremygurr** [comment]
yes I'm happy to discussing rune effects. That is the part that needs the most help. Like I said, it's just a very rough draft so far. 

I can see that high rune caster wins would be very difficult as things currently stand, but keep in mind that the rune curses are being done soon to prepare the way for some significant buffing changes coming after, particularly for casters. So things will get a little too hard for a while, but steadily become easier, as I complete the extra scrolls and potions (amplification and inversion alone will be game changers). There is a big spell rework I want to do which will greatly enhance how powerful a magic user can get (see the future file).

Also keep in mind that spellcasters are already significantly more powerful in CA than in vanilla crawl, since spell power for all spells is capped at 200 (allowing ensorcelled hibernation, for example, to be useful even against some late game enemies).

**Comment by Sandman25DCSS** [comment]
I still amnesia Ensorcelled hibernation before late game, Confuse is much better since it does not require Ice Magic and Mass Confusion does not have power cap in standard crawl either.
Spellcasters rely on retreating to restore mana, it is hard to get reliable method to channel mana in 3 rune game, the suggested rune curses will make pure casters impossible (similar to pure summoners even without rune curses, I guess you don't like pure anything :))

When rune effect triggers? When the rune is picked up? Then players will clear levels but will not pick up any runes. Runelock of vaults will force players to pick a rune and it the rune will not be determined by "how hard is the level" but by "how annoying/hard the rune curse is".
Here are my concerns with specific rune curses:

> decaying: player rots at a steady rate.

It is annoying. I still remember how I complained about Ghoul who lost 30% HP early game because it was unable to find any monsters to fight.

> barnacled: player movement is slowed down by 0.2

The rune will probably be picked up last, after previous 14 runes.

> gossamer: traps more common and more deadly

I hope you remember my attitude towards traps. Random deaths without player fault are unavoidable evil (the lesser evil) in current crawl but must not become a design goal.

> slimy: player randomly mutates, good or bad 50/50, like a bad version of evolve

Similar to traps, dying to teleportitis or berserkitis (by the way is their a ay to prevent berserk in no hunger crawl?) without player faults

> dark: player vision is reduced to 3, but can still sense the position of monsters in normal LOS, just can't id them until they are closer

Reduced vision is a bonus, see darkness and nightalker

**Comment by G0rg0** [comment]
I see the allure for rune curses as it adds a lot of flavor, but I have to agree with Sandman. It sounds remarkably painful to play, particularly for casters.

I would also say that while casters might be more powerful on paper, in practice, they seem weaker than in vanilla.

**Comment by jeremygurr** [comment]
I like players to be able to be pure summoners or any other kind of caster. If that can't happen, then adjustments are needed. The last time I played summoner in this fork I found it quite easy to be a pure summoner, at least in the early stages of the game. 

Yes, players would clear levels and not pick up runes, and then go back to pick up the runes they want. That seems a bit lame. What if, once a player enters a rune branch, they can't leave until they have the rune? The abyss of course would be an exception, but you can't clear out the monsters from the abyss anyway. That's pretty vicious, but I think it could be balanced to make things fun. The idea of clearing out all but the last floor of a branch, and coming back later to clear out the last floor I think is both spoilery and unnecessarily tedious. I think once you are powerful enough to clear out the top floors of a branch, you also ought to be able to handle the bottom. 

Another possibility: until you get the rune from a rune branch, re-entering the branch repopulates it with monsters. So it wouldn't accomplish anything to just clear the floor and come back later for the rune. 

Yes I would expect players to factor in the rune curse when deciding which runes to get. I think that adds to the depth. But the rune curses should not be so unbalanced as to make all players want to make the same decisions for a 3 rune game in every case. There should be situations where one rune curse is much better than another, and that should change based on the character. 

player rot: This can be made to happen slowly. I'm not sure how it applies to ghouls though since ghouls have to eat to restore rot. In this case, you will need a source of healing to cure rot. Maybe healing potions just fully cure rot every time. So you allow as much rot as you can before drinking a potion to remove it all. Definitely ups the challenge considerably, which is the objective here. I don't think it's too tedious to drink a healing potion every 10 floors or so to remove rot. 

slowed movement: is slower movement really so bad? It could just be 0.1. Players already move slower than monsters in normal exertion mode. And by late game most players will have a source of hasting or at least swiftness, and many other escaping options. Nagas and chei characters play the entire game that way. And of course they would probably pick this rune up first. Which is exactly the kind of differentiation I'm going for. 

traps: yeah this is one of the lamest. Please send any suggestions you can think of for replacements :)

random mutations: How is this any different than monsters that mutate the player? Maybe they could be random temporary mutations if that's too much. Or a player can only have 1 mutation at a time caused by this rune, so when another mutation comes, the previous one is replaced. 

reduced vision: no it will not be a bonus in this case to have vision reduced. By the point I would implement this, there would be two LOS parameters for a player. There would be the normal LOS that other monsters use to determine if they can see the player, which would not change in this case, then there is what is actually revealed to the player on their screen, which is what is being reduced here. This will be used to both increase player vision distance (for the glow mutation), and to provide a negative darkness effect (like for this rune). 

Player's WILL be able to notice that there is a monster there, even if it's out of their vision range, if it's within the normal LOS that is used by the monster for awareness. So it will still stop autoexplore when a monster is close enough to become aware of the player, even if it's not yet in visual range. But a player will just see a question mark where the monster is, until it enters their vision range. And of course the monster can't be targetted until it is within visual range.

**Comment by jeremygurr** [comment]
the point of the rune curses isn't to add flavor, but to increase the challenge of late game. 

Why do people mutate themselves to add challenge to the game, because it is too easy for them? Does the game become painful because they mutated themselves? What is the difference between making a game more fun because it is more challenging, and making a game painful because it is more challenging?

I think all challenge in a game should put the player in a position to come up with better solutions, until the challenge is no longer a significant challenge. That's what I'm going for here. If that isn't happening, then it needs to change. I will be adding many items/abilities/etc. to the game that will enable the player to get more powerful and have more ways of dealing with some of these super hard challenges.

**Comment by Sandman25DCSS** [comment]
Have you thought about rune curses being double-edged thing? Something like:
•serpentine: stairs randomly teleport nearer monsters ~every 50 turns. Bonus: ability to teleport for 5% HP
•decaying: player rots at a steady rate. Bonus: potion of curing heals double number of HP.
•barnacled: player movement is slowed down by 0.1. Bonus: ability to invoke blink for free (gives Exh status)
•gossamer: EV -3. Bonus: ability to temporary get EV+5 and AC+5 (gives Exh status)
•abyssal: monster spawning increased significantly.  Bonus: killing enough monsters generates an escape hatch to previous floor (does not work in Pan and probably Hell)
•silver: ascending stairs sometimes moves player to a cluster of monsters on same floor instead. Bonus: player can seal doors like Vault Warden does
•slimy: player randomly mutates, good or bad 50/50, like a bad version of evolve. Bonus: player gets 50% chance to block incoming mutations (neqoxecs, cacodemons etc.)
•dark: player vision is reduced to 3, but can still sense the position of monsters in normal LOS, just can't id them until they are closer. I think it won't work.
•glowing: contamination doesn't auto decay. Bonus: Can be cured completely by drinking any potion.
•fiery: monster damage increased by 25%. Bonus: player can activate "peace" mode when all damage (both player and monster) is decreased by 40%, gives Exh status
•magical: power of monster spells increased. Bonus: power of player spells can be increased, gives Exh status
•demonic: when a hit does a lot of damage to a monster, there is a chance that a demon will spawn. Bonus: ability to have a chance to turn those spawned demons friendly, gives Exh status, does not affect already spawned demons
•golden: gold in los has a chance to distract player, causing their attack to miss, or their spell to fail. Higher chance of gold being dropped by monsters. Bonus: ability to get Augmentation based on gold in view, gives Exh status
•iron: walls are all indestructable, but paths to stairs are always guaranteed. By the way it would mean Slimy rune should be obtained first. Bonus: ability to activate walls reflection effect (all fire/cold/missile attacks are reflected like Lightning Bolt)
•icy: ice walls or tunnels randomly appear. Can be melted with heat. Never close to player. Bonus: ability to activate ice magic enhancer, gives Exh status
•obsidian: walls and trees adjacent to the player occasionally come to life and attack the player, but don't move. When the player moves away from a living wall, it eventually fades back into a normal wall. Bonus: ability to get 10 AC, gives Exh status
•bone: when a monster dies, there is a 1/3 chance it comes back as a zombie or demon. Bonus: ability to get Necromancy wizardry, gives Exh status

As you can see I tried to use Exh status, so player cannot spam those abilities and must choose what is the most important for him currently. I am sure my ideas are bad, it's just a 10 minutes example ;)

I think we discussed limited view in "discuss future plans", I believe player should not be killed without seeing monsters and Fire Storm still should be useful in extended.
It's possible to avoid mutations as stealthy character with ranged attack.

**Comment by G0rg0** [comment]
I guess the main thing is balance. I see extended as pretty challenging already, so making it more challenging seems unnecessary, but that's without seeing any of the buffs you have planned.

I would hate to see a situation where it's technically _possible_ to overcome these sorts of challenges but the path to doing so becomes tedious / not-fun. I know you are against tedium as well, so I am not too worried, just wanted to toss that out there.

**Comment by jeremygurr** [comment]
I like the idea of a double edged runes. I'd like to avoid the exhausted status and use stamina instead if possible, since they really represent the same concept. It could either cost a fixed amount of stamina, or a % of total stamina, or even reduce max stamina temporarily. I'm also okay with other limiters such as Breath, and possibly using mp instead of sp, or even other resources to limit usage of abilities. I also want to make sure that a limiter is actually needed. I think many of the ability limiters in the game are meaningless (2 MP for one thing, minor hunger cost for another), and they would be better not to have the cost at all. 

Yeah I don't intend to allow monsters to attack the player before the player can be aware that they are there. But you are saying you want to be able identify them immediately. Okay, maybe instead the darkness mechanism allows you to identify the monster at a distance, but you can't target them until they are inside your real-los. So really it would become an attack range reduction, instead of a visibility reduction. I think that would still work. And it doesn't have to be 3. It could just be 1 or 2. That would still make quite an impact.

**Comment by Sandman25DCSS** [comment]
Stamina as limit could work too as long as there is no way to quickly get full stamina back. I am not used to stamina so I suggested an old and boring Exh :)
Another idea for dark:
Penalty: 20% chance to miss any monster (even with no miss tools like Firestorm/airstrike/freeze etc.). It is checked before usual accuracy is applied.
Bonus: activate Light ability, gives sInv and Corona to everything including player. Could be useful for bolt of inaccuracy, summons who don't see Inv.

**Comment by jeremygurr** [comment]
Although it is true that extended is pretty challenging for most of us already, there are many players who can consistently achieve 15 rune wins in vanilla crawl, and even multi-zig attacks on top of it. For those players, crawl becomes much less interesting, because they've already plumbed it's depths. I want to add substantial depth. I want it to be a significant achievement to pull off a 5 rune win on normal mode. It would be amazing on hard mode. And it should be epic to pull off a 10 rune win. 15 rune wins should be the stuff of legends. It's actually quite difficult to pull that off in a game, but I think we have a real chance here. This greater depth will make room for more fun tools I'll be adding, without making everything feel OP for advanced players. 

Yeah we won't end up with tedium. At least not after sufficient playtesting and tweaking. I don't doubt this will be a substantial challenge to pull off without being tedious, but I think it's doable. 

I think I'll start with 5 rune curses, the other runes will be untouched for now (although they will still have the 10% hp boost on monsters per rune collected). This is a relatively easy feature to implement, especially compared with stamina and bringing back old branches. I still have a bit of work to do on stamina, but once the stamina change is playing well and feeling good, rune curses are probably next. 

So most common 5 runes: 3 lair runes, vaults, and abyss?

**Comment by jeremygurr** [comment]
As far as spellcasters being weaker, that might be a bug. It's possible I don't have spellpower scaling properly, and that in vanilla crawl you actually end up with more spell power at a given point in the game than here. If so, just tell me so I can adjust it.

**Comment by Sandman25DCSS** [comment]
Spellcasters are weaker not because of lower spell power,  they are weaker because eventually they will run out of MP, that's a sure thing provided some rune curses teleport player to monsters or create monsters near player.

**Comment by Sandman25DCSS** [comment]
By the way I am not sure HP bonus to monsters should apply to easy mode, the mode should stay easy IMHO

**Comment by jeremygurr** [comment]
spellcasters are mp bound, but removing the level 27 cap allows them to gain more mp than they would have in an vanilla game. They can continue to train spellcasting and the mp pool will continue to increase because of XL gains. Also there will be many new items coming before long (replication scrolls, amp scrolls, inversion), that will offer many more options for dealing with those tough sitations and for helping a player to manage their scarce consumables better. Add to this the spellcasting rework that will be coming later, that will make magic costs per spell a variable which goes down with skill, then we will likely be seeing more of a need to nerf spellcasters than to buff them.

The HP bonus is pretty important for making the extended game a challenge. It of course can be reduced for easy mode (say 5% increase per rune), but the limiter should still be there. Maybe hard mode would be 15% per rune. 

An easy mode player should have a relatively easy time winning a 3 rune game. But I don't think that means they should be able to pull off a 15 rune win without some major challenges. I'm really going for this: an easy mode 15 rune win should be **harder** than a vanilla crawl 15 rune win is now. **Significantly harder**. While a 3 rune win on easy should be dramatically easier than a vanilla 3 rune win. Veterans should be able to manage a 15 rune win on easy mode, but it should still push them pretty hard. A 15 rune win in normal or hard should be pretty much unheard it. Years may pass before anyone accomplishes it. See the vision?

**Comment by Sandman25DCSS** [comment]
I see, makes sense then.

**Comment by jeremygurr** [comment]
would a monster damage increase per rune be better than the hp increase? It would just be nice to have something based on the total number of runes that you have. 

Or maybe both a hp and damage increase. And it doesn't even have to be that much. If necessary we could even do 1% easy, 2% normal, 3% hard for each. Although I would like to try 5/10/15 first and see if that is too bad.

**Comment by jeremygurr** [comment]
Ok I have a version deployed with the 5/10/15 % increase for hp and damage. Now I need someone to attempt a 15 rune win...

**Comment by Sandman25DCSS** [comment]
I think damage increase is even worse idea than HP increase. Being one-shot during autoexplore is not fun and I think it will happen too often for Sp with 10+ runes.
I will try 15 runes on hard... if I survive the first 3 :).

**Comment by jeremygurr** [comment]
I will eventually implement the one-shot death removal feature. It's not hard, I just need to take care of this save game issue before I bork any more of them. 

I can always tone down the % if it is too much. Right now I'm trying 5/10/15, but maybe 2/4/6 would work just fine. It just takes some experimenting to find out.

**Comment by Sandman25DCSS** [comment]
I think rune curse should activate when player reaches rune floor i.e. Snake 4, Tomb 3, Gehenna 7, Abyss 3+ etc. The only exception should be done for Pan.
I was cast into Abyss 5 from Crypt and decided to pick up the rune despite it increases monster HP, I think picking up the rune should be always a good idea (except for demonic rune in Pan maybe).

**Comment by jeremygurr** [comment]
Then players would just clear everything but the bottom floor of each branch. What if, instead, a floor of a rune branch refilled with monsters each time the player reentered the branch without the rune? And refill monsters don't give exp or drop items. 

I think this could be applied to abyss and pan too, to avoid scumming. Abyss monsters never give exp or drop items. Abyss floors don't give exp on floor based exp modes. Items don't spawn on floor of the abyss except for initial visit to the floor. Same for Pan. Only monsters on rune floors in pan give exp/ drop items, since those only spawn once. Overall item spawning and exp gain can be adjusted to compensate for this.

**Comment by Sandman25DCSS** [comment]
Rune floor is much more dangerous than previous floors, many players often clear Vault 1-4, then Crypt/Elf/Depths and only after that enter Vaults 5.
I think refilling previous floors with monsters will not do much, they will be popcorn, at least this is my impression from my orb run game, I basically didn't care about merfolks in Shoals at XL 27.
Abyss scumming is rarely a good idea, it has very dangerous monsters, monsters who mutate player and problems with teleport/blinking. Scumming Lair for XP or Orc/Elf for items is much better probably, now that player cannot starve.
I don't think we should nerf scumming Pan, it scares me a lot already. Even if I have no runes before entering Pan, I can have at least 3 runes (if I refuse to take demonic rune before taking unique floors) and then fight Lom Lobon who has 145% HP and deals more damage. Hell Sentinels with 160% HP, Fiends with 160% HP etc., I don't think it can be called scumming ;)

Also I am worried a bit about Zot 5. Previously it was optimal to clear it after getting all 15 runes, now I am not sure. Orbs of Fire with 325% HP can be unkillable for many characters but clearing Zot 5 with 3 runes can make extended impossible due to mutations.

**Comment by jeremygurr** [comment]
That popcorn becomes quite a bit more threatening after you've picked up a few runes, and eliminates the motivation to leave runes behind. 

Players are not going to ever see that happen though. Just the fact that it would happen is enough of a deterrent to prevent a player from wanting to leave behind the rune to make the game easier. 

I don't really like that optimal play means saving the last floor of each branch for later, then requiring the player to come back when stronger to finish it off. That adds forced tedium. Not a big deal, but still not the best. The bottom floor of a branch should only be marginally more difficult than the upper floors. Ideally I would want to either make the upper floors more dangerous, or the lower floor less dangerous, or a little of both. 

That being said I'm not against just making the rune floor respawn on reentry. Sounds potentially too easy though. But if that's enough of a deterrent, I guess it would be simpler than what I'm proposing.

**Comment by Sandman25DCSS** [comment]
Leaving last floor for later is optimal in standard crawl too. With weak character I often switch to Orc 1 after Lair 7, to Swamp 1 after Snake 3, to Depths after Elf 2, famous Vaults 5 etc.

**Comment by jeremygurr** [comment]
Yeah but is that a good thing? Wouldn't you rather that lair 8 be the next
most difficult thing to tackle, since you are already at lair 7? Instead of
jumping all around the map? That is also a spoilery thing for new players.
Most players would assume that after you beat one level, the place to go is
the following level. What other rpg ever does this?

**Comment by Sandman25DCSS** [comment]
I don't know. Weak character pay piety and food to switch to another branch, strong characters clear last floor too. Also we have Lair branches of roughly similar difficulty, first level of one Lair branch will always be much easier than last level of another Lair branch.

**Comment by jeremygurr** [comment]
fair enough. Okay last floor will get new monsters if you enter it again without the rune.

**Comment by Sandman25DCSS** [comment]
Also I think rune curse should activate after entering rune level, otherwise player will pick it up only after killing all monsters.

**Comment by alexjurkiewicz** [comment]
> fair enough. Okay last floor will get new monsters if you enter it again without the rune.

This will prevent stairdancing and off-level healing. Perhaps you could add a grace period (say 1000 turns).

**Comment by yrmvgh** [comment]
How much would it mess with Core Crawl Gameplay As Intended, if some branches sealed you in until you got the rune?

I guess we already have branches where the rune disappears if you leave, which is a little more elegant - you CAN retreat, but you give up a rune forever.

**Comment by Sandman25DCSS** [comment]
I kind of start to see that the idea really works. Player HP and hence standard/challenging/nightmare mode is the most important factor (15 runes on standard is easier than 3 runes on nightmare), then extra HP from runes makes extended really hard on any level. For example, Tomb is almost unbeatable without rTorment which is relatively easy to get on standard but extremely hard on nightmare.

**Comment by yrmvgh** [comment]
Because spider/snake and shoal/swamp are pretty much required for the 3 rune game, I think it's especially important to ensure those 2 / 4 runes are equally good or bad for almost any character type.

Barnacled having -.1 mv is a buff to chei players, nerf to everyone else, though power mode felids and spriggans may not notice.

**Comment by jeremygurr** [comment]
Yeah I fully expect that some rune curses will be much harder on some combos than others, and I think that's a good thing, since the optimal path to picking up runes will not be exactly the same for each character. 

I'd like to have the difficulty of all of the rune branches to be more or less even, where they get naturally more difficult as you gain runes. Right now we have dramatic differences in difficulty between hells, abyss, vaults, tomb, shoals, etc. This is necessary in vanilla crawl, otherwise the game gets very boring after you've picked up the first few runes. 

But in a game with rune curses, I think it's interesting to be able to choose any rune branch to tackle first, based on your resistances / items / character build. Then as you get a few more runes, the rest of the rune branches naturally become much harder. So if you tackled hell first, it wouldn't be too bad. Then if you put off spider until last, it would be the nastiest most difficult thing you've ever seen. Contrast this with the current situation where the optimal path is almost always to get 2 lair runes, vaults, abyss or slime, ... you get the picture. 

With rune curses implemented on top of the old vanilla crawl difficulty progression, it may be a bit too much. 

I'm also considering putting all 5 lair runes in every game. The randomness currently present is definitely fun, but think for a moment of what it would be like if you had 5 of them in lair. Now with vanilla crawl this would make it much easier to get the needed 3 runes for Zot. But with rune curses that isn't the case. After you've picked up a couple of runes, the 3rd rune will be at least as difficult as having to go to vaults or abyss in vanilla crawl to retrieve your third, even if your 3rd rune comes from snake or spider. We could even make special vaults that appear based on how many runes the player has, adding more dimension to the branches. So taking on shoals as your first rune may look quite different than taking on shoals as your 15th. This is mostly for flavor though, most of the difficulty is provided by the rune curses themselves. 

This of course would mean there are 17 runes in the game. I'm actually aiming for 20. I'd like a nice round number :) . I think we could possibly put a rune in elf (since it follows the same sub-branch pattern used by tomb and the lair branches), and definitely one in fortress. That puts us at 19. Well, unless anyone can think of another good place for a rune, maybe the orb itself becomes the 20th, completing the set :) ... one rune to rule them all haha... The orb of course already has it's own curse.

**Comment by jeremygurr** [comment]
Ok so at the moment I'm thinking of this for rune curses: the curse activates upon entering the first floor of a rune bearing branch. But you don't get the new ability that goes with the rune until you actually pick up the rune. This way people don't just leave the last floor of each branch until they've cleared out everything else in the game.

Of course for pan, the curse activates when you enter a rune floor. If you leave the floor without the rune, the curse stays with you.... muahahaha.... well maybe not. That's probably unnecessarily mean.

**Comment by yrmvgh** [comment]
Rune curses are hard to care / have strong feelings about, since I still don't feel like I've mastered 3 rune vanilla games yet. But they sound like a great longevity / replay value feature.

I mean, I can get high AC melee-chars with a strong god (preferably not Trog) through the game, but that's the easiest route. Once I've done it using bad gods, stabbers, casters, shooters, and evokers (or some combinations of the above), then I'll really be more interested in rune curses and able to form solid opinions.

If the 2 easy lair runes become "an option" rather than "the default path", that makes it a loss less of a problem if some of the runes' downsides are nbd! Cool idea!

**Comment by jeremygurr** [comment]
Yeah I rarely get past 3 runes myself, but the way it's going I think it will even have a big impact on those that never play the extended game. If we flatten out the branch difficulty curve a bit (not completely), then you may head into pan or vaults for your first rune instead of always one of the lair branches. And even in 3 rune games, the effects of those runes will impact the player.

**Comment by Sandman25DCSS** [comment]
Well, rune curses make different rune order optimal, basically you are interested in the hardest rune you can currently get safely. Something like Shoals->Tomb->Vaults->Hells->Pan->Abyss->Slime->rest of Lair. Optional branches like Elf/Dwarf/Crypt/Forest/Depths should be cleared with just 1 rune, maybe even 0.
I was trying to compare CA fork to vanilla fork so I still used default vanilla rune order but even then I picked Cocytus first, Gehennah next and the easiest Dis/Tar last because I didn't want to fight Antaeous or Briemstone Fiends with maxed HP. Basically after 1-2 runes player usually gets less extra HP from every rune than monsters do so it does not make any sense to postpone them. If you can clear Cocytus, do it now or it will become harder the longer you wait.

**Comment by jeremygurr** [comment]
Yeah but that's partially because hell is meant to be so much more difficult than the other branches, which shouldn't ideally be the case in the rune curse world. Just forcing a different optimal path isn't what I'm going for. I want the player to have to think about the best path for each different build, what items you happen to have come across in a game, etc. 

I'm some rune perks could potentially be a one time ability or gift as opposed to active abilities. Maybe one rune could allow you to gain one intrinsic resistance, and another allows you to choose a mutation from a list of 10 powerful fixed mutations. Maybe another gives you the one-time ability to significantly add to your mutation resistance. Another could allow you to choose one artifact item from a list of 10. Another could offer a selection of powerful scrolls you could choose from. 

So when you have available rune perks, a single ability shows up in the ability menu allowing you to use the perk, then asking which perk you want to use.

**Comment by jeremygurr** [comment]
But I don't actually want to nerf hell much. Instead I want to add some challenge vaults to the easier branches that only show up after you have some number of runes, to partially level the playing field. 

Also, there won't be any more optional branches, since elf and dwarf will have a rune, and forest must be passed through to get to elf, and crypt must be passed through to get to tomb.

**Comment by Sandman25DCSS** [comment]
I wouldn't care about mutations in extended. No matter what your character and tactics are, you will be mutated a lot just because Cacodemons and Neqoxecs have huge HP and refuse to die to 3 bolts from Hellfire arbalet or +6 exec. axe. If some rune gives protection from mutation (even partially), that would greatly encourage going for it first so it should be one of the hardest runes, unless it is on a random Pan floor like Mnoleg's one so you get it when you are basically done with the area which has the most of mutations.

**Comment by Sandman25DCSS** [comment]
By the way activating rune curse right after entering first floor of a branch can be cruel to new players, they can enter Slime or Shoals just to check what's inside.

**Comment by jeremygurr** [comment]
Here's some ideas for rune perks (remember: these are one time only):
- lightning form: can move at 0.1 speed for 100 movements, which is really just 10 turns at lightning speed. 
- one time scroll selection: choose from a list of 10 powerful scrolls, which are always the same (not randomized). Some may be multiples. 
- one time potion selection: Probably would include 10 cure mutations, 10 healing, 50 cure, 10 haste, etc.
- one time artifact jewelry selection: random list of 10 jewelry artifacts
- one time artifact armour selection
- one time artifact weapon selection
- free learning of any spell in the game, without costing spell slot points
- free teleport to depth 5, no matter where you are
- select one mutation to remove
- select one mutation to gain out of a fixed (not random) list of 10 powerful mutations
- select one permanent resistance to gain: strong mutation resistance, strong magic resistance, rF++, rC++, rN++, rPois, rElec
- teleport all monsters in LOS to D:1 surrounding the dungeon exit. So you don't have to deal with them now, but you eventually will... could save your life though if things get out of hand
- select one dna booster: good dna (more likely to get good muts when mutated), clean dna (more likely to lose muts when mutated), resilient dna (more likely to keep good muts when losing muts during mutation)

We could not allow any more rune perks to be used once the player gets the orb, if we find the ascension too easy, but I'm not sure that will be the case. I think even with the rune perks the ascension with the rune curses will be very difficult, even with just the 3 runes. 

So there are 4 mutation related perks in the list so far. Maybe all 4 should be from hell runes?

**Comment by jeremygurr** [comment]
Before entering a rune branch, the game will prompt "Entering this branch will activate a rune curse, which in this case will increase monster hp and damage by 2%, and slow your movement by 0.2. Are you sure you want to enter?"

**Comment by jeremygurr** [comment]
Keep in mind that several buffs are coming for the player which will make some of the harder branches easier than they would otherwise be, to help level the playing field. So it will get harder first (too hard, really), but then it will slowly get easier, taking it from "impossibly difficult" to just "insane but possible".

**Comment by Sandman25DCSS** [comment]
Those ideas are interesting, I like them.

**Comment by jeremygurr** [comment]
So optimal path may be to clear all the non-rune branches first (Dungeon, lair, orc, depths, crypt, forest, zigs, zot), then deal with the runes. This seems okay to me.

**Comment by jeremygurr** [comment]
Since you can be sent to the abyss unintentionally, maybe you only get the rune curse if you go deeper into the abyss on purpose, or pick up the rune, whichever comes first.

**Comment by yrmvgh** [comment]
> I want the player to have to think about the best path for each different build, what items you happen to have come across in a game, etc.

That is going to be pretty opaque to the unspoiled, or to people checking out the fork anew. Sounds like one of those things where the player is going to be looking at the death screen, thinking they did everything right and confused as to why they are dead.

Perhaps picking up a rune could force the { screen to come up, and that screen can have detailed info on what you're getting yourself into if you go pick up more runes/curses/perks

**Comment by Sandman25DCSS** [comment]
Probably it could be fun to make some rune curses depend on number of runes when that specific rune is picked up. I mean that penalty/bonus for barnacled rune could differ between games where it was 1st rune obtained and games where it was 2nd rune obtained and so on. For example:
•one time artifact jewelry selection: random list of N jewelry artifacts where N is number of runes after picking up that specific rune
•lightning form: can move at 0.1 speed for 100 movements, which is really just 10 turns at lightning speed. Can be used N times where N is number of runes after picking up that specific rune. Cannot be used on levels containing runes (otherwise it is too easy to steal a rune)
•select N mutations to remove. N is number of runes after picking up that specific rune

**Comment by jeremygurr** [comment]
yrmvgh: Just before you enter a rune branch, the game will clearly spell out the rune curse consequences, and offer the player the option to continue or turn back. I believe this solves the issue you are pointing out. Correct me if I'm wrong. 

Sandman: I like that. Rune count powered bonuses. 

Oh, another possibliity occurred to me based on what you were saying: What if each rune perk gains a "charge" each time you pick up another rune. It's effectively the same as what you are saying for some of the perks, except that you don't have to wait for as long as possible to maximize the gain. 

For example, you pick up the lightning rune first, and so can use lightning form once. Then you pick up the artifact jewelry rune. Now you have 2 charges on the lightning rune (assuming you haven't used it yet), and 1 on the artifact jewelry rune. Then you pick up the mutation rune, so you have 3 lightning charges, 2 artifact jewelry charges, and 1 mutation charge. So it makes a huge difference which order you get the runes in. Of course the more powerful perks will be found on the more difficult to obtain runes.

**Comment by jeremygurr** [comment]
Maybe we don't need the specific curse part. Maybe the rune curse is only the hp and damage increase (which is pretty significant by itself). Simplicity is good when possible.

**Comment by Sandman25DCSS** [comment]
Yes, charging can be even better, it is like using (16-N) instead of N for 15 runes games. I guess we can pick the best approach for specific curse.

**Comment by jeremygurr** [comment]
I'll start by just adding the perks, and leave out the specific rune curses, until we find that we need them.

**Comment by Sandman25DCSS** [comment]
I suspect HP/damage alone won't be enough to make 15 runes almost impossible on nightmare, uncannily stealthy DD of Makhleb with Necromutation and AC 60+/EV 30+ does not care much about damage.
http://crawl.homedns.org/morgue/Sandman25/morgue-Sandman25-20160426-032919.txt

**Comment by jeremygurr** [comment]
Ok we'll come up with some fun and challenging curses then. Another fun idea I'm wanting to do for orb run is, after picking up the orb, remove one out of every 3 up stairs for challenge mode. For nightmare mode, 2 out of 3 will be removed. 

I'd really like to make the specific curse part have a coolness factor if possible, although that takes a lot more creativity. Maybe something along the lines of you randomly change gods every few thousand turns, but keep your piety and don't have any wrath. This throws a curveball at the player, but doesn't make it impossible. Or maybe another curse could randomly transfer charges from one wand to another. You never lose charges, but plans could definitely be messed up.

**Comment by Sandman25DCSS** [comment]
I would not like random gods change, it should not be mandatory. It's ok for new species or new god only IMHO. Pure caster joining Trog or Fire Storm user joining Dith is not fun, at least for some players.
Personally I never join Xom, for example.

**Comment by jeremygurr** [comment]
The random god change is just one specific rune curse. And I don't think it should make the character completely unplayable. The rune would only change a player to Trog if they had no spells memorized, which late in the game is almost never. Maybe the game would check to see if the player is already violating the god's conduct, and not switch to those. But Xom would always be a possibility, since there's nothing he really doesn't like. I want to use the same logic for the ecumenical altars.

**Comment by Sandman25DCSS** [comment]
Ok, I guess player can always win with 14 runes and pretend like it's all 15 if they don't like xom-like curses.

**Comment by jeremygurr** [comment]
You have to admit it would definitely make things interesting to occasionally have your god swapped out for another. Maybe we could have it alternate between a random god and your chosen god, so it's not as bad. So after 500 turns, it takes you from Sif to Fed. You deal with Fed for the next 500 turns, but then it gives you Sif back for the next 500 turns. Then you get Xom and things really get tough, but then you are back to Sif again. I think the flavor and difficulty is exactly what I'm going for with the specifc curses. Now I have to find 18 other unique curses that are as good as this one :)

I've no doubt that some runes will just be avoided by some players / combos. That's part of the point. The curse should heavily influence which runes are picked up and in which order. Unlike in vanilla crawl, it shouldn't be a normal thing for players to aim for a 15 rune win (or a 19 rune win in circus). Typically players will aim for 5 or 6. Then when they can do that consistently, then they try for 7, maybe 8. It should be a very slow and challenging progression to get up past 10 runes. And that's just in "standard" mode :) ... It should be quite difficult even for super experienced players to pull off a 19 rune win in **standard** mode. So skipping one or two that are particularly annoying would be quite common and reasonable. 

Even on standard mode with 2% increases in hp and dam per rune, after you've picked up 19 runes, all monsters you have to deal with in the ascension will have 81% more hp and damage, which is nothing to sneeze at :). This is on top of 19 specific rune curses the player has to deal with ... should be quite formidable even for a veteran.

Thinking about the ascension and rune curses, I think when the player enters Zot they should be locked in the branch until they pick up the orb. So they shouldn't enter Zot until they are committed to the ascension. This is to avoid the player going into Zot before picking up any runes, and clearing it out, and then coming back after they've picked up the 19 runes. So much better if Zot could be waiting in full force ...

**Comment by jeremygurr** [comment]
No single rune curse is meant to be crushingly difficult by itself. In fact, by itself, it shouldn't be too big of a deal. Just like adding 2% to hp and damage isn't a big deal. But stacked up, they become dramatically more difficult.

**Comment by jeremygurr** [comment]
Oh, in nightmare mode after picking up 19 runes, monster hp will be 474% above their base (almost 6x original values). Yeah it's _supposed_ to be a nightmare. That's why it's called that :P

You're going to need your 60 ac gargoyle at level 35 to even have a ghost of a chance. And all of the other player buffs I've added and will add.

**Comment by Sandman25DCSS** [comment]
No, I have to admit I am among those players who would go for 14 runes only. Forcing a Formicid in CPA of Makhleb to switch to Vehumet or Chei with Invo 0 is not interesting to me. Forcing an uncannily stealthy Enchanter of Dith with Invisibility switch to TSO or Qazlal is not interesting to me. Keeping multiple weapons in inventory because suddenly your demon trident of anything or executioner's axe of venom can stop working is not interesting to me.

**Comment by Sandman25DCSS** [comment]
I like Zot lock idea except it punishes new players who are unaware that Orbs of Fire and Electric Golems exist.

**Comment by jeremygurr** [comment]
Yeah there will be a big warning sign at Zot entrance: "Warning there are horrible creatures here beyond anything you have ever seen before, and once you enter, you cannot escape unless you are carrying the orb. So most likely, you will die. Are you sure you want to enter?"

I see your point with god changes not working. The challenge needs to be doable and interesting, but not overwhelming. Hmm... back to the drawing board.

**Comment by Sandman25DCSS** [comment]
If you want a challenge related to gods, it can be something like Wrath card from deck of punishment. 

> The Wrath card invokes the wrath of a randomly chosen god against the player

**Comment by jeremygurr** [comment]
That's definitely a possibility, but not the ideal, because I'd like it to more significantly affect the player's strategies over a longer period of time than a single battle, although I suppose some wrath accomplishes this. But I may not be able to come up with anything better.

**Comment by Sandman25DCSS** [comment]
Well, rune curses are in good state now, I died during Orb run with 6 runes from more than half HP (62/119) to a pack of popcorn, I just didn't expect it :( The level was tough, Panlord and Hell Sentinel caused me to teleport 3 times.
http://crawl.homedns.org/morgue/Sandman25/morgue-Sandman25-20160502-234131.txt
But at least I learned that golden rune can be obtained without necromutation :)
I will try some better species, something like Naga. Deep Elf is too fragile on nightmare, glass cannon just does not work.

**Comment by Sandman25DCSS** [comment]
Uh, was the damage from monsters increased without changing log messages? According to my log I lost less than 62 HP during last turn.

> The balrug gestures at you.
> The fireball explodes!
> The fireball engulfs the sun demon. (0)
> The sun demon completely resists.
> The fireball engulfs you! (36)
> You resist.
> The fireball engulfs the sun demon. (0)
> The sun demon completely resists.
> The fireball engulfs your fire vortex. (0)
> Your fire vortex completely resists.
> The sun demon hits you (15)!
> Damage (15) was greater than 30% of your hp (24)!!!
> You are engulfed in flames. (5)
> You resist.
> The sun demon hits you (10)!
> You are engulfed in flames. (4)
> You resist.
> You die...

So I got 12 from Fireball (rF++ for 36), 15 from melee attack, 5 from flames, 10 from melee and 4 from flames. That's 46, not 66.

**Comment by jeremygurr** [comment]
That 36 damage displayed is after all ac and resists are taken out. All damage should show the exact damage taken from hp. I can't guarantee that all of it is being displayed, however, because crawl has damage messages scattered all over the place. I just found a couple new ones today. 

I actually see 70 damage there, which would have dropped you to -4, which seems to be consistent what your morgue file says. 

It is a little misleading how it shows the damage and then says "You resist." Maybe I can figure out a clearer way for that to be communicated.

**Comment by jeremygurr** [comment]
Oh I misread a part. I thought you were saying you started with 66, but you started with 62, which means it's displaying 4 damage too much. Hmm... I wonder what's going on there.

**Comment by Sandman25DCSS** [comment]
I don't think it is damage after AC for Fireball, as you see I got 36 damage from Balrug and Balrug's Fireball deals 3d23 damage, that's 1d23 vs my rF++.

**Comment by jeremygurr** [comment]
first rune curse implemented for elven runes. Eventually I'll get to the rune perks also.
